docs/design/keyboard-shortcuts.md

```diff
@@ -0,0 +1,184 @@
+1. In the JavaScript file, use calls of the [Office.actions.associate](/javascript/api/office/office.actions#associate) API to map a function to each action name that you used in the JSON file. To begin add the following to the file. Note the following about this code:
+
+    - The first parameter is one of the action names from the JSON file.
+    - The second parameter is the function that runs when a user presses the key combination that is mapped to the action name in the JSON file.
+
+    ```javascript
+    Office.actions.associate('-- acton name goes here--', function () {
+
+    });
+    ```
+
+1. To continue the example, use `'SHOWTASKPANE'` as the first parameter.
+1. For the body of the function, use the [Office.addin.showTaskpane](/javascript/api/office/office.addin.md#showastaskpane--) method to open the add-in's task pane. When you are done, the code should look like the following:
+
+    ```javascript
+    Office.actions.associate('SHOWTASKPANE', function () {
+        return Office.addin.showAsTaskpane()
+            .then(function () {
+                return;
+            })
+            .catch(function (error) {
+                return error.code;
+            });
+    });
+    ```
+
+1. Add a second call of `Office.actions.associate` function to map the `HIDETASKPANE` action to a function that calls [Office.addin.hide](/javascript/api/office/office.addin.md#hide--). The following is an example:
+
+    ```javascript
+    Office.actions.associate('HIDETASKPANE', function () {
+        return Office.addin.hide()
+            .then(function () {
+                return;
+            })
+            .catch(function (error) {
+                return error.code;
+            });
+    });
+    ```
+
+1. Sideload the add-in and toggle the visibility of the task pane by pressing Ctrl+Shift+UpArrow and Ctrl+Shift+DownArrow.
+
+An example of a simple add-in that uses several custom keyboard shortcuts is at [shortcut-sample-revision1](https://github.com/OfficeDev/testing-assets/tree/master/addins/shortcut-sample-revision1).
+
+## Using key combinations that are already used by Office or another add-in
+
+During the preview period, there is no system for determining what happens when a user presses a key combination that is registered by an add-in and also by Office or by another add-in. Behavior is undefined.
+
+Currently, there is no workaround when two or more add-ins have registered the same keyboard shortcut, but you can minimize conflicts with Excel with these good practices:
+
+- Use only keyboard shortcuts with the following patterns in your add-in.
+
+    - **Alt-*n***, where *n* is a numeral from 1 to 9.
+    - **Ctrl+Shift+Alt+*n***, where *n* is a numeral from 1 to 9.
+
+- If you need more keyboard shortcuts, check the [list of Excel keyboard shortcuts](https://support.microsoft.com/office/keyboard-shortcuts-in-excel-1798d9d5-842a-42b8-9c99-9b7213f0040f), and avoid using any of them in your add-in.
```

Comment:
Actual list is different. Please refer to the following link. `c_mapStringToModifierKey` contains all the possible modifiers. `c_mapStringToVirtualKey` contains the possible other key. Note that any character in A-Z, a-z, 0-9 ranges is also possible for one other key.

https://dev.azure.com/office/_search?action=contents&text=ShortcutMappingTableManager.cpp&type=code&lp=custom-Collection&filters=ProjectFilters%7BOfficeSourceDepot%7DRepositoryFilters%7BOffice_devmain%7DBranchFilters%7Boffice_devmain%7D&pageSize=25&result=DefaultCollection%2FOfficeSourceDepot%2FOffice_devmain%2FGBOffice_devmain%2Fosfclient%2Fosfactions%2FShortcutMappingTableManager.cpp